src/content/blog/fastify.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: "Revamping our API pt. 1: Moving from express to Fastify"
+description: How switching to Fastify let us embrace runtime and compile-time types
+pubDate: June 28, 2024
+author: Tom MacWright
+---
+import NpmChart from './fastify/NpmChart.astro';
+
+The JavaScript ecosystem has changed a lot in the last 15 years.
+We've gone from CommonJS to ESM, from browserify to webpack to vite,
+from mocha to vitest, from the request module to using web-standard
+fetch everywhere. The constant reinvention can
+be exciting or annoying from one day to the next.
+
+### The success of Express
+
+In the midst of all this churn, [express][express] has persevered.
+Express is a web framework for Node.js, introduced
+by [TJ Holowaychuk][tj] in 2009. Like a lot
+of early JavaScript modules and many of TJ's projects, it took inspiration
+from the land of Ruby - in particular, [Sinatra][sinatra], which
+was established in 2007.
+
+```js title="express-example.js"
+// This is a tiny example of using
+// express. Notice how there's almost no boilerplate
+// involved in creating a tiny web server.
+const app = require('express')();
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+});
+app.listen(3000);
+```
+
+Express is a beautifully light abstraction: a
+'hello world' example is barely over 10 lines. Compared to the boilerplate
+required to do the same in Ruby on Rails or a Java server at the time, it
+was a breath of fresh air.
+
+<NpmChart />
+
+Express became the default for new JavaScript projects, and has stayed
+there. In a typical week, express is downloaded from NPM 30 million
+times. It's a roaring success in every way.
+
+### A need for types
+
+Express is an _expressive_ framework: like a classic Ruby or
+JavaScript framework, it lets you color outside the lines by using
+loosely-defined objects and types. Query string parameters in express
+become [freeform objects](https://expressjs.com/en/4x/api.html#req.query).
+The same goes for the bodies of requests, and the kinds of responses
+you can send: it's all a bit ad-hoc by default.
+
+But the vibes have shifted: we write [TypeScript][ts] now,
+and everyone's excited about validator modules like [Zod](https://zod.dev/)
+and [TypeBox](https://github.com/sinclairzx81/typebox). Interfaces
+are less ad-hoc and more declared, tested, and strictly typed.
+
+Modern tooling is written to take full advantage of types
+and schemas. With [tRPC](https://trpc.io/), we're writing backend
+endpoints and calling them from auto-generated, auto-typed functions
+on the frontend. Other backends are embracing [OpenAPI][openapi]
+(née Swagger) to generate clients, test suites, and more.
```

Comment:
I had to look up née lol. The joke I'd make is "the artist formerly known as Swagger"